dvc/cache.py

```diff
@@ -82,25 +78,28 @@ def _get_remote(self, config, name):
                 [cache]
                 ssh = dvc-storage
 
-        This method resolves the name under the cache section into the
-        correct Remote instance.
+        This method creates a cached property, containing cache named `name`:
 
-        Args:
-            config (dict): The cache section on the config file
-            name (str): Name of the section we are interested in to retrieve
+            self.config == {'ssh': 'dvc-storage'}
+            self.ssh  # a RemoteSSH instance
+        """
 
-        Returns:
-            remote (dvc.Remote): Remote instance that the section is referring.
-                None when there's no remote with that name.
+        def getter(self):
+            from dvc.remote import Remote
 
-        Example:
-            >>> _get_remote(config={'ssh': 'dvc-storage'}, name='ssh')
-        """
-        from dvc.remote import Remote
+            remote = self.config.get(name)
+            if not remote:
+                return None
+
+            return Remote(self.repo, name=remote)
 
-        remote = config.get(name)
+        getter.__name__ = name
+        return cached_property(getter)
 
-        if not remote:
```

Comment:
If remote cache is not configured (usually none or 1 of them are), then remotes are not loaded and neither their dep libs. Am I missing something? Also it would be great to have before vs after time comparison to see if such unobvious complication is worth it or not.

Reply:
But if it's configured then it will be loaded for each operation even local one.